cylc/flow/install_plugins/log_vc_info.py

```diff
@@ -275,41 +302,40 @@ def _parse_svn_info(info_text: str) -> Dict[str, Any]:
     return ret
 
 
-def get_diff(vcs: str, path: Union[Path, str]) -> Optional[str]:
-    """Return the diff of uncommitted changes for a repository.
+def write_diff(
+    vcs: str, repo_path: Union[Path, str], run_dir: Union[Path, str]
+) -> Path:
+    """Get and write the diff of uncommitted changes for a repository to the
+    workflow's vcs log dir.
 
     Args:
         vcs: The version control system.
-        path: The path to the repo.
-    """
-    args_ = DIFF_COMMANDS[vcs]
-    if Path(path).is_absolute():
-        args_.append(str(path))
-    else:
-        args_.append(str(Path().cwd() / path))
-
-    try:
-        diff = _run_cmd(vcs, args_, cwd=path)
-    except (VCSNotInstalledError, VCSMissingBaseError):
-        return None
-    header = (
-        "# Auto-generated diff of uncommitted changes in the Cylc "
-        "workflow repository:\n"
-        f"#   {path}")
-    return f"{header}\n{diff}"
-
-
-def write_diff(diff: str, run_dir: Union[Path, str]) -> None:
-    """Write a diff to the workflow's vcs log dir.
-
-    Args:
-        diff: The diff.
+        repo_path: The path to the repo.
         run_dir: The workflow run directory.
+
+    Returns the path to diff file.
     """
+    args = DIFF_COMMANDS[vcs]
+    args.append(
+        str(repo_path) if Path(repo_path).is_absolute() else
+        str(Path().cwd() / repo_path)
+    )
+
     diff_file = Path(run_dir, LOG_VERSION_DIR, DIFF_FILENAME)
     diff_file.parent.mkdir(exist_ok=True)
-    with open(diff_file, 'w') as f:
-        f.write(diff)
+
+    with open(diff_file, 'a') as f:
+        f.write(
+            "# Auto-generated diff of uncommitted changes in the Cylc "
+            "workflow repository:\n"
+            f"#   {repo_path}\n"
+        )
+        f.flush()
+        try:
+            _run_cmd(vcs, args, repo_path, stdout=f)
+        except VCSMissingBaseError as exc:
+            f.write(f"# No diff - {exc}")
```

Comment:
I think this is more "could not extract diff" than "no diff"? Should it say "ERROR"?

Reply:
Not so much an error, it's basically "Git diff does not produce an output when there is no base commit"